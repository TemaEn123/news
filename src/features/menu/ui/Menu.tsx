import React from 'react';
import { categories } from '@/shared/constants';
import { NavLink, useParams } from 'react-router-dom';
import styles from './styles.module.scss';

const Menu = () => {
  const params = useParams();

  return (
    <nav className={styles.menu}>
      <ul>
        {categories.map((cat) => (
          <li key={cat}>
            <NavLink
              className={`${styles.menu__link} ${params.cat === cat ? styles.menu__link_active : null} white`}
              to={`category/${cat}`}
            >
              {cat}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Menu;
