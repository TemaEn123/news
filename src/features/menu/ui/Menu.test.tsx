import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Menu from './Menu'; // Убедитесь, что путь к компоненту правильный

describe('Menu Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Router>
        <Menu />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
