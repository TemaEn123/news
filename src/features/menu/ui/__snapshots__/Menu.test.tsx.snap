// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="menu"
  >
    <ul>
      <li>
        <a
          class="menu__link null white"
          href="/category/business"
        >
          business
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/entertainment"
        >
          entertainment
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/general"
        >
          general
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/health"
        >
          health
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/science"
        >
          science
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/sports"
        >
          sports
        </a>
      </li>
      <li>
        <a
          class="menu__link null white"
          href="/category/technology"
        >
          technology
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
